tests/backends/base/test_creation.py

```diff
@@ -101,3 +105,38 @@ def test_circular_reference(self):
         obj_ref = ObjectReference.objects.get()
         self.assertEqual(obj.obj_ref, obj_ref)
         self.assertEqual(obj_ref.obj, obj)
+
+    def test_serialize_circular_ref_with_natural_keys(self):
+        # serialize_db_to_string and deserialize_db_from_string handle circular and self-references for models with
+        # natural keys.
+
+        a = CircularRefThingA.objects.create(key="X")
+        b = CircularRefThingB.objects.create(key="Y", other_thing=a)
+        a.other_thing = b
+        a.save()
+
+        t1 = SelfRefThing.objects.create(key="X")
+        t2 = SelfRefThing.objects.create(key="Y", other_thing=t1)
```

Comment:
I moved test strictly related with ticket-26552 to a separate PR #12651.

Reply:
Left a review there. Do you want me to update this PR accordingly, or will you?

Reply:
Updated, thanks.